diff --git a/node_modules/@notifee/react-native/src/types/Trigger.ts b/node_modules/@notifee/react-native/src/types/Trigger.ts
index 6c0760a..e386538 100644
--- a/node_modules/@notifee/react-native/src/types/Trigger.ts
+++ b/node_modules/@notifee/react-native/src/types/Trigger.ts
@@ -34,6 +34,19 @@ export interface TimestampTrigger {
   alarmManager?: boolean | TimestampTriggerAlarmManager | undefined;
 }
 
+/**
+ * An interface representing the different alarm types which can be used with `TimestampTrigger.alarmManager.type`.
+ *
+ * View the [Triggers](/react-native/docs/triggers) documentation to learn more.
+ */
+export enum AlarmType {
+  SET,
+  SET_AND_ALLOW_WHILE_IDLE,
+  SET_EXACT,
+  SET_EXACT_AND_ALLOW_WHILE_IDLE,
+  SET_ALARM_CLOCK,
+}
+
 /**
  * Interface to specify additional options for the AlarmManager which can be used with `TimestampTrigger.alarmManager`.
  *
@@ -43,11 +56,17 @@ export interface TimestampTrigger {
  */
 export interface TimestampTriggerAlarmManager {
   /**
+   * @deprecated use `type` instead
+   * -----
+   *
    * Sets whether your trigger notification should be displayed even when the system is in low-power idle modes.
    *
    * Defaults to `false`.
    */
   allowWhileIdle?: boolean;
+
+  /** The type of alarm set by alarm manager of android */
+  type?: AlarmType;
 }
 
 /**
