diff --git a/node_modules/react-native-system-setting/android/src/main/java/com/ninty/system/setting/SystemSetting.java b/node_modules/react-native-system-setting/android/src/main/java/com/ninty/system/setting/SystemSetting.java
index dedd524..548ddae 100644
--- a/node_modules/react-native-system-setting/android/src/main/java/com/ninty/system/setting/SystemSetting.java
+++ b/node_modules/react-native-system-setting/android/src/main/java/com/ninty/system/setting/SystemSetting.java
@@ -532,12 +532,10 @@ public class SystemSetting extends ReactContextBaseJavaModule implements Activit
 
     @Override
     public void onHostResume() {
-        registerVolumeReceiver();
     }
 
     @Override
     public void onHostPause() {
-        unregisterVolumeReceiver();
     }
 
     @Override
@@ -597,4 +595,23 @@ public class SystemSetting extends ReactContextBaseJavaModule implements Activit
             }
         }
     }
+
+    private int listenerCount = 0;
+
+    // Required for rn built in EventEmitter Calls.
+    @ReactMethod
+    public void addListener(String eventName) {
+        if (eventName.equals("EventVolume")) {
+            listenerCount++;
+            registerVolumeReceiver();
+        }
+    }
+
+    @ReactMethod
+    public void removeListeners(Integer count) {
+        listenerCount -= count;
+        if (listenerCount == 0) {
+            unregisterVolumeReceiver();
+        }
+    }
 }
